### PR TITLE
Refactor sidecar command, add configs

### LINKED ISSUE #240: Refactor sidecar module
* [x] inject only available configs
 * [x] split `util` module into multiple modules (separation of concerns)

### PR DESCRIPTION
**Pull request by AMecea**
fixes #240

### CODE REVIEW

File: pkg/sidecar/appconf.go [hunk 1]
```diff
@@ -14,65 +14,60 @@ See the License for the specific language governing permissions and
...
 	var dynCFG, utilityCFG, clientCFG *ini.File
 	var err error
 
-	if dynCFG, err = getDynamicConfigs(util.GetServerID(), reportHost); err != nil {
+	if dynCFG, err = getDynamicConfigs(cfg.ServerID, reportHost); err != nil {
 		return fmt.Errorf("failed to get dynamic configs: %s", err)
 	}
 
-	if err = os.Mkdir(util.ConfDPath, os.FileMode(0755)); err != nil {
+	if err = os.Mkdir(confDPath, os.FileMode(0755)); err != nil {
```
Comment (calind) [inline pkg/sidecar/appconf.go]: Maybe move this before `dynCFG` related code.

File: pkg/sidecar/appconf.go [hunk 2]
```diff
@@ -14,65 +14,60 @@ See the License for the specific language governing permissions and
...
 		if !os.IsExist(err) {
-			return fmt.Errorf("error mkdir %s/conf.d: %s", util.ConfigDir, err)
+			return fmt.Errorf("error mkdir %s/conf.d: %s", configDir, err)
 		}
 	}
-	if err = dynCFG.SaveTo(util.ConfDPath + "/10-dynamic.cnf"); err != nil {
+	if err = dynCFG.SaveTo(confDPath + "/10-dynamic.cnf"); err != nil {
```
Comment (calind) [inline pkg/sidecar/appconf.go]: Let's call this something more explicit, like `serverIdentity`. `dynCFG` and `10-dynamic.cnf` is pretty general.

File: pkg/sidecar/apphelper.go [hunk 3]
```diff
@@ -14,86 +14,81 @@ See the License for the specific language governing permissions and
...
 	// wait for mysql to be ready
-	if err := waitForMysqlReady(); err != nil {
+	if err := waitForMysqlReady(cfg); err != nil {
 		return fmt.Errorf("mysql is not ready, err: %s", err)
 	}
 
 	// deactivate super read only
 	log.Info("temporary disable SUPER_READ_ONLY")
-	if err := util.RunQuery("SET GLOBAL READ_ONLY = 1; SET GLOBAL SUPER_READ_ONLY = 0;"); err != nil {
+	if err := runQuery(cfg, "SET GLOBAL READ_ONLY = 1; SET GLOBAL SUPER_READ_ONLY = 0;"); err != nil {
```
Comment (calind) [inline pkg/sidecar/apphelper.go]: Can we have the defer pattern here:
```go
enableSuperReadOnly, err := disableSuperReadOnly()
defer enableSuperReadOnly()
```

File: pkg/sidecar/appconf.go [hunk 4]
```diff
@@ -14,65 +14,60 @@ See the License for the specific language governing permissions and
...
 		return fmt.Errorf("failed to save configs: %s", err)
 	}
-	if utilityCFG, err = getUtilityUserConfigs(util.UtilityUser, uPass); err != nil {
+	if utilityCFG, err = getUtilityUserConfigs(utilityUser, uPass); err != nil {
 		return fmt.Errorf("failed to configure utility user: %s", err)
 	}
-	if err = utilityCFG.SaveTo(util.ConfDPath + "/10-utility-user.cnf"); err != nil {
+	if err = utilityCFG.SaveTo(confDPath + "/10-utility-user.cnf"); err != nil {
```
Comment (calind) [inline pkg/sidecar/appconf.go]: Use `path.Join` for joining paths.
```suggestion
	if err = utilityCFG.SaveTo(path.Join(confDPath, "10-utility-user.cnf")); err != nil {
```

File: pkg/sidecar/apphelper.go [hunk 5]
```diff
@@ -235,11 +231,51 @@ func markConfigurationDone() error {
...
+
+// getGTIDFrom parse the content from xtrabackup_binlog_info file passed as
+// io.Reader and extracts the GTID.
+func getGTIDFrom(reader io.Reader) (string, error) {
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(bufio.ScanWords)
+
+	count := 0
+	gtid := ""
+	for scanner.Scan() {
```
Comment (calind) [inline pkg/sidecar/apphelper.go]:
```suggestion
	for i:=0; scanner.Scan(); i++ {
```

File: pkg/sidecar/apphelper.go [hunk 6]
```diff
@@ -235,11 +231,51 @@ func markConfigurationDone() error {
...
+		if count == 2 {
+			gtid = scanner.Text()
+		}
+		count++
+	}
+
+	if err := scanner.Err(); err != nil {
+		return "", err
+	} else if len(gtid) == 0 {
+		return "", fmt.Errorf("failed to read GTID reached EOF")
```
Comment (calind) [inline pkg/sidecar/apphelper.go]: nit:
```suggestion
		return "", io.EOF
```

File: pkg/sidecar/configs.go [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+	// MasterNode represents the master role for MySQL server
+	MasterNode NodeRole = "master"
+	// SlaveNode represents the slave role for MySQL server
+	SlaveNode NodeRole = "slave"
+)
+
+// Config contains information related with the pod.
+type Config struct {
+	// Stop represents the shutdown channel
+	Stop <-chan struct{}
```
Comment (calind) [inline pkg/sidecar/configs.go]: We should pass the stop channel to the function which requires it.

File: pkg/sidecar/apphelper.go (hunk 3 above)
Comment (AMecea) [reply pkg/sidecar/apphelper.go]: This PR does not introduce `disableSuperReadOnly` function. Only in #243 this function is added.

**Pull request merged**